### ISSUE TITLE
REST API?

### ISSUE THREAD

**Issue by brylie** [q2a/question2answer#398]
We are considering how to integrate data from question2answer into a custom application. Is there a REST API that we can us? What are some options for data integration?

**Comment by brylie** [comment]
@kyyberi, have you done some work to this effect?

**Comment by phy25** [comment]
You can use q2a's functions directly in other PHP code: http://docs.question2answer.org/code/functions/

**Comment by brylie** [comment]
Our application is written in JavaScript, so a REST API would be important.

**Comment by svivian** [comment]
There is no REST API included as yet, but it is something I've been considering.

I don't think anyone had created a full API plugin, but there are a few possibilities listed [here](http://www.question2answer.org/qa/32740/is-there-any-rest-or-soap-api-is-available) that may help you.

**Comment by kyyberi** [comment]
Yes, I created API PoC some years ago (see above link). Did not have time to work with it after switching employer.

**Comment by brylie** [comment]
Thanks @kyyberi. What resources would it take to get Q2A core support for a REST API? E.g. merging this REST package into core.

**Comment by yerbol89** [comment]
Good question. @svivian please update and add in new 1.8 version. REST API for Android and iOS app.

**Comment by svivian** [comment]
I did ask [this question about a Kickstarter](http://www.question2answer.org/qa/47757/how-would-you-feel-about-kickstarter-q2a-plugin-development) over at Q2A a little while back. So if there are several people willing to donate I can spend a week (at least) of work time developing an API plugin. Otherwise, it's something that will get done eventually but not super high priority right now.

I believe @patricksavalle from Mobbr was interested in something similar too. Not sure if they solved all their requirements already though.

**Comment by brylie** [comment]
@svivian are there any crowdfunding sites you would consider using that allow contributions for individual Github issues?

**Comment by svivian** [comment]
@brylie I'm pretty open to anything that helps make Q2A better! Did you have any crowdfunding sites in mind?

**Comment by brylie** [comment]
We have bid on [Bountysource](https://www.bountysource.com/) issues in the past.

**Comment by isabasan** [comment]
Is there any progress on the REST API?

**Comment by Calinou** [comment]
Hi, has anyone made a plugin recently that adds a read-only REST API? I'd like to display a list of questions matching a tag search at the bottom of a static HTML page. While this might be doable using an iframe, it'd be cleaner to use a REST API so I don't have to manually exclude the sidebar and stuff like that.
